From 42eeaa2e477d79976d90bf22c5aee3f80b88a2a3 Mon Sep 17 00:00:00 2001
From: Erhan Bilgili <erhan.bilgili@gmail.com>
Date: Sun, 2 Aug 2026 10:46:47 +0200
Subject: [PATCH] winewayland: Avoid double-counting server decorations.

window_rect_from_visible() expands the configured visible rect by the
current frame extents. Do not add the same server-decoration size before
that conversion.

(cherry picked from commit 42eeaa2e477d79976d90bf22c5aee3f80b88a2a3)

Source: https://github.com/nanomatters/wine-salkim/commit/42eeaa2e477d79976d90bf22c5aee3f80b88a2a3
---
 dlls/winewayland.drv/window.c | 12 +-----------
 1 file changed, 1 insertion(+), 11 deletions(-)

diff --git a/dlls/winewayland.drv/window.c b/dlls/winewayland.drv/window.c
index 9d6baf83739..6f0c49ddcd1 100644
--- a/dlls/winewayland.drv/window.c
+++ b/dlls/winewayland.drv/window.c
@@ -922,7 +922,7 @@ void WAYLAND_WindowPosChanged(HWND hwnd, HWND insert_after, HWND owner_hint, UIN
 static void wayland_configure_window(HWND hwnd)
 {
     struct wayland_surface *surface;
-    INT width, height, window_width, window_height, offset_x, offset_y;
+    INT width, height, window_width, window_height;
     UINT flags = 0;
     UINT state_cmd = 0;
     uint32_t state;
@@ -1013,19 +1013,9 @@ static void wayland_configure_window(HWND hwnd)
 
     window_width = width;
     window_height = height;
-    offset_x = ((surface->window.rect.left - surface->window.window_rect.left) +
-                (surface->window.window_rect.right - surface->window.rect.right));
-    offset_y = ((surface->window.rect.top - surface->window.window_rect.top) +
-                (surface->window.window_rect.bottom - surface->window.rect.bottom));
 
     flags |= SWP_NOACTIVATE | SWP_NOZORDER | SWP_NOOWNERZORDER | SWP_NOMOVE;
     if (window_width == 0 || window_height == 0) flags |= SWP_NOSIZE;
-    /* avoid any behavior differences when server side decorations is disabled */
-    else if (surface->processing.decor == ZXDG_TOPLEVEL_DECORATION_V1_MODE_SERVER_SIDE)
-    {
-        window_height += offset_y;
-        window_width += offset_x;
-    }
     SetRect(&rect, 0, 0, window_width, window_height);
     OffsetRect(&rect, data->rects.window.left, data->rects.window.top);
     if (!IsRectEmpty(&rect)) rect = window_rect_from_visible(&data->rects, rect);
-- 
2.55.0

